packages/template/-private/signature.d.ts

```diff
@@ -41,21 +41,24 @@ export type ComponentSignatureBlocks<S> = S extends { Blocks: infer Blocks }
         ? { Params: { Positional: Blocks[Block] } }
         : Blocks[Block];
     }
-  : EmptyObject;
+  : {};
 
 /** Given a component signature `S`, get back the `Element` type. */
-export type ComponentSignatureElement<S> = S extends { Element: infer Element } ? Element : null;
-
-// These shenanigans are necessary to get TS to report when named args
-// are passed to a signature that doesn't expect any, because `{}` is
-// special-cased in the type system not to trigger EPC.
```

Comment:
This goes away now by way of handling it through arity rather than through parameter type, right?

Reply:
Yep, exactly. (And actually, even if that weren't the case, the `NamedArgs` wrapper type we employ now would end up triggering EPC anyway)